### DISCUSSION TITLE
Is it possible to display the name of the month in blogs?

### DISCUSSION THREAD

**Discussion by ghost** [datenstrom/community#51] [Ask a question] [answered]
Thus, the name of the month in the snippet "blog archive" displays correctly, must
<? php setlocale (LC_TIME, "de_DE"); ?>
be set.

More: https://php.net/manual/en/timezones.php

Could not do something in the config.ini?

Greeting
Peter

**Comment by markseuffert** [comment]
Hi Peter!

Yes, it's a good workaround... if the date is not in your native language.

How to solve time and date output in general. Just last week I thought about it, when I saw that the blogarchive uses a special way to format dates. My suggestion is maybe not very traditional, maybe not what traditional PHP developers would do. From the feedback I got via email there are a many different PHP systems out there, they are not always up to date, web hostings don't behave all the same way. So instead of relying on a system locale, how about the following: Lets put date strings into Yellows language files. For example `January, February, March` and so on. As a result Yellow could create month names via the normal API. This would give developers one function to use and would work in all CMS features. In the future we could also make fancy things like blog postings that show `Today` instead of the full date.

What do others think about this?

**Comment by ghost** [comment]
Hi Mark

I find a super idea!
The solution I've only done because I found no other way.

Thank You
Peter

Gesendet von Windows Mail

Von: Mark Seuffert
Gesendet: ‎Montag‎, ‎26‎. ‎Januar‎ ‎2015 ‎14‎:‎05
An: markseu/yellowcms
Cc: Peter Nagl

Hi Peter!

Yes, it's a good workaround... if the date is not in your native language.

How to solve time and date output in general. Just last week I thought about it, when I saw that the blogarchive uses a special way to format dates. My suggestion is maybe not very traditional, maybe not what traditional PHP developers would do. From the feedback I got via email there are a many different PHP systems out there, they are not always up to date, web hostings don't behave all the same way. So instead of relying on a system locale, how about the following: Lets put date strings into Yellows language files. For example January, February, March and so on. As a result Yellow could create month names via the normal API. This would give developers one function to use and would work in all CMS features. In the future we could also make fancy things like blog postings that show Today instead of the full date.

What do others think about this?

**Comment by markseuffert** [comment]
Thanks, then this wasn't a crazy idea after all. Let's ~~make~~ put it on the TODO list.

**Comment by ghost** [comment]
No not at all.

You could, for example, the following Insert in the language file:

$days = array("Sonntag", "Montag", "Dienstag", "Mittwoch", "Donnerstag", "Freitag", "Samstag");

$months = array(1=>"Januar",
                2=>"Februar",
                3=>"M&auml;rz",
                4=>"April",
                5=>"Mai",
                6=>"Juni",
                7=>"Juli",
                8=>"August",
                9=>"September",
                10=>"Oktober",
                11=>"November",
                12=>"Dezember");

Thank You
Peter

Gesendet von Windows Mail

Von: Mark Seuffert
Gesendet: ‎Montag‎, ‎26‎. ‎Januar‎ ‎2015 ‎14‎:‎53
An: markseu/yellowcms
Cc: Peter Nagl

Thanks, then this wasn't a crazy idea after all. Let's make put it on the TODO list.

**Comment by markseuffert** [comment, marked as answer]
Months and weekdays are now supported. Let me know if it's working for you.

**Comment by ghost** [comment]
And how do I use the change?

Greeting
Peter

**Comment by markseuffert** [comment]
Make a backup of your `system` folder, update to latest Yellow and extensions.

**Comment by markseuffert** [comment]
Should be working now.
